roadside-forms-frontend/frontend_web_app/src/components/common/Header/Header.js

```diff
@@ -66,10 +68,12 @@ export const Header = () => {
             }
             setUserRoleData(data);
             setuserAdminInfo(data.some( role => role['role_name'] === 'administrator' ))
+            setLoginCompleted(true);
           }
           else {
             setUserRoleData([]);
             console.log(resp.data.error);
+            setLoginCompleted(true);
```

Comment:
why is this set to true in both cases?

Reply:
LoginCompleted atom is introduced because other atoms such as userRoles are not loading fast enough and due to that user is getting redirected to request access page even though they do have roles, but as we are loading roles in header, and redirecting user in requestAccess both are running in parellel which leads to no userRoles found, so to fix it have to set this atom to true so that we can pass that process of setting user roles doesn't matter what outcome we get. once we have that login completed setup, we display access page based on that or by pass requestaccess to display dashboard.